File: sample-ecommerce/iosApp/SparrowDeskStore/CartManager.swift
import SwiftUI

class CartManager: ObservableObject {
    static let shared = CartManager()

    @Published private(set) var items: [Product] = []

    var count: Int { items.count }
    var total: Double { items.reduce(0) { $0 + $1.price } }
    var formattedTotal: String { String(format: "$%.2f", total) }

    func add(_ product: Product) {
        items.append(product)
    }

    func remove(at index: Int) {
        items.remove(at: index)
    }

    func clear() {
        items.removeAll()
    }
}

File: sample-ecommerce/androidApp/src/main/java/com/sparrowdesk/ecommerce/CartManager.kt
package com.sparrowdesk.ecommerce

object CartManager {
    private val items = mutableListOf<Product>()

    fun add(product: Product) {
        items.add(product)
    }

    fun remove(product: Product) {
        items.remove(product)
    }

    fun getItems(): List<Product> = items.toList()

    fun getCount(): Int = items.size

    fun getTotal(): Double = items.sumOf { it.price }

    val formattedTotal: String get() = "$${String.format("%.2f", getTotal())}"

    fun clear() {
        items.clear()
    }
}
